lib/foreman_rh_cloud/engine.rb

```diff
@@ -5,11 +5,19 @@ module ForemanRhCloud
   class Engine < ::Rails::Engine
     engine_name 'foreman_rh_cloud'
 
-    def self.register_scheduled_task(task_class, cronline)
+    def self.register_scheduled_task(task_class, cronline, env_cronline = nil)
       ForemanTasks::RecurringLogic.transaction(isolation: :serializable) do
-        return if ForemanTasks::RecurringLogic.joins(:tasks)
-                  .merge(ForemanTasks::Task.where(label: task_class.name))
-                  .exists?
+        existing_logic = ForemanTasks::RecurringLogic.joins(:tasks)
+                         .merge(ForemanTasks::Task.where(label: task_class.name))
+                         .where(state: 'active')
+                         .first
```

Comment:
Only one at a time, because of the serializable transaction above.

As for re-entry, now it should not matter. Unless the ENV is set explicitly, the cron line would not change.